### PR TITLE
Fix multi process collision by initializing pid variable in master process

### PR DESCRIPTION
**Pull request by vvkuznetsov**

### CODE REVIEW

File: prometheus_client/core.py [hunk 1]
```diff
@@ -395,6 +395,14 @@ def close(self):
 def _MultiProcessValue(_pidFunc=os.getpid):
     files = {}
     values = []
+
+    # 1. use dict instead of nonlocal keyword to support python 2.7
```
Comment (brian-brazil) [inline prometheus_client/core.py]: This comment isn't adding much to someone familiar with Python, I'd remove it.

File: prometheus_client/core.py [hunk 2]
```diff
@@ -413,22 +421,24 @@ def __init__(self, typ, metric_name, name, labelnames, labelvalues, multiprocess
...
+                        os.environ['prometheus_multiproc_dir'], '{0}_{1}.db'.format(file_prefix, pid['value']))
                 files[file_prefix] = _MmapedDict(filename)
             self._file = files[file_prefix]
             self._key = json.dumps((metric_name, name, labelnames, labelvalues))
             self._value = self._file.read_value(self._key)
 
         def __check_for_pid_change(self):
-            if self._pid != _pidFunc():
+            actual_pid = _pidFunc()
+
```
Comment (brian-brazil) [inline prometheus_client/core.py]: you can remove this blank line

Comment (vvkuznetsov) [reply prometheus_client/core.py]: ok

File: prometheus_client/core.py (hunk 1 above)
Comment (vvkuznetsov) [reply prometheus_client/core.py]: agree

File: prometheus_client/core.py [hunk 3]
```diff
@@ -395,6 +395,13 @@ def close(self):
 def _MultiProcessValue(_pidFunc=os.getpid):
     files = {}
     values = []
+
+    # pid needs to be evaluated here in a master process
```
Comment (brian-brazil) [inline prometheus_client/core.py]: That's usage dependant, not all relevant systems have masters. I think it'll be fine not to have this comment at all.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**